fix: skip sqlite filename warning when connection is a function

innerdvations opened the pull request:
## Summary

When `connection` is a function, Knex stores it as a `connectionConfigProvider` and does not resolve settings synchronously in the constructor. The sqlite constructor was checking `config.connection.filename` directly, which is always `undefined` for a function — so users with a valid dynamic config got a false warning.

Guard the existing constructor check with `typeof config.connection !== 'function'`. Static misconfiguration (e.g. `connection: {}` without `filename`) still warns at construction.

## Test plan

- [x] `test/integration2/dialects/sqlite.spec.js` — `connection.filename warning` block for both `sqlite3` and `better-sqlite3`
- [x] Targeted mocha grep for the new tests — 6 passing

innerdvations committed:
fix: defer sqlite filename warning until connection settings resolve

When connection is a function, Knex does not resolve connectionSettings
until the pool first acquires a connection. Warn for missing filename
only once settings are available: early for static misconfiguration,
and in acquireRawConnection after a provider resolves.

Also validate in better-sqlite3's overridden acquireRawConnection and
add integration coverage for both sqlite drivers.

tgriesser commented:
I think the GPT generated code here is overkill, the ideal fix is just to just change:

```ts
if (config.connection && config.connection.filename === undefined) {
```
 
 to
 
```ts
if (config.connection && typeof config.connection !== 'function' && config.connection.filename === undefined) {
```

since what we're basically assuming is that if you've specified a function for connection, we don't want/need to warn.

innerdvations committed:
fix: skip sqlite filename warning when connection is a function

Address review feedback: guard the constructor check with
typeof config.connection !== 'function' instead of deferring
validation to acquireRawConnection.

innerdvations commented:
Ahh, that's what I had originally 😆 I figured if we defer we could still validate that it does eventually resolve to a filename. Either way works for me. Updated now to the simpler version.
